components/engine_rocks/src/write_batch.rs

```diff
@@ -124,3 +153,192 @@ impl Mutable for RocksWriteBatch {
             .map_err(Error::Engine)
     }
 }
+
+/// `RocksWriteBatchVec` is for method `multi_batch_write` of RocksDB, which splits a large WriteBatch
+/// into many smaller ones and then any thread could help to deal with these small WriteBatch when it
+/// is calling `AwaitState` and wait to become leader of WriteGroup. `multi_batch_write` will perform
+/// much better than traditional `pipelined_write` when TiKV writes very large data into RocksDB. We
+/// will remove this feature when `unordered_write` of RocksDB becomes more stable and becomes compatible
+/// with Titan.
+pub struct RocksWriteBatchVec {
```

Comment:
Can this be implemented in pingcap/rocksdb?

Reply:
It could do this. I implemented it in TiKV just because it is convenient to adjust the methods of this structure.

Reply:
But the interface is stable now, isn't it?

Reply:
Emmm. Yes, maybe we can move it into rocksdb next PR...